### PR TITLE
Committed create project new changes here--progressing on Save part

### PR DESCRIPTION
**Pull request by 212040-AnnMariaJenson**

### CODE REVIEW

File: package-lock.json
Comment (sourcecoursepoc) [inline package-lock.json:1]: don't commit package-lock.json

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 1]
```diff
@@ -41,24 +41,34 @@ const GroupsModalBox: React.FC<MyModalProps> = ({
   onExport,
   onCreatePipeline,
 }) => {
+  const [tableData,setTableData]=useState<TableProps>();
   const [groupModalBoxTreeView, setGroupModalBoxTreeView] = useState(true);
   const [displayAttributeSection, setDisplayAttributeSection] = useState(false);
   const [schema, setSchema] = useState<string | null>(null);
   const [saveModalVisible, setSaveModalVisible] = useState(false);
   const [selectedNodeDetails, setSelectedNodeDetails] = useState([]);
+  console.log(selectedNodeDetails, "groupmodalBox--selectedNodeDetails");
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove console.log

File: src/components/ComposePage/GroupsPage/ModalBox/attributeDisplayBox.tsx [hunk 2]
```diff
@@ -11,7 +12,7 @@ interface MyComponentProps {
   attribute: string;
   icon: ReactNode;
   uid: string;
-  handleRemove?: (uid: string) => void;
+  handleRemove: (uid: string) => void;
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/attributeDisplayBox.tsx:15]: is handleRemove madatory?

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 3]
```diff
@@ -44,13 +44,23 @@ const GroupsModalBox: React.FC<MyModalProps> = ({
...
   const dispatch = useDispatch();
   const database = useSelector(getDataBaseSelector);
+  console.log(database, "databasedatabasedatabase");
+  const tablesAndColumns = database.map((node) => node.tables);
+  useEffect(() => {
+    setTableData(tablesAndColumns);
+  }, []);
+  
+  console.log(tableData,"tableData")
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove consoles

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 4]
```diff
@@ -44,13 +44,23 @@ const GroupsModalBox: React.FC<MyModalProps> = ({
...
+  console.log(tablesAndColumns,"tablesAndColumns")
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove console

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 5]
```diff
@@ -69,6 +79,8 @@ const GroupsModalBox: React.FC<MyModalProps> = ({
     },[fetchProjectSchemaInfo]);
 
   useEffect(() => {
+    dispatch(fetchDataBaseRequest());
+    // dispatch(fetchGroupDataRequest());
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove commented line

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 6]
```diff
@@ -87,8 +99,8 @@ const GroupsModalBox: React.FC<MyModalProps> = ({
         ]);
       }
     }
-  }, [selectGroupdataData,attrName]);
-console.log("lastIndiceslastIndices",lastIndices)
+  }, [selectGroupdataData, attrName]);
+  console.log("lastIndiceslastIndices", lastIndices);
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove console

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 7]
```diff
@@ -131,28 +143,40 @@ console.log("lastIndiceslastIndices",lastIndices)
...
-  function getValues(attributeName:string,selectedDataType:string){
-        setAttrName(attributeName)
-        setDatatype(selectedDataType)
-        selectGroupdataData.slice(-1)[0] = {name: attributeName, type: selectedDataType}
-        lastIndices[lastIndices.length-1] = {name: attributeName, type: selectedDataType}
-       }
+  function getValues(attributeName: string, selectedDataType: string) {
+    setAttrName(attributeName);
+    setDatatype(selectedDataType);
+    selectGroupdataData.slice(-1)[0] = {
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: use condition operator (?)

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 8]
```diff
@@ -131,28 +143,40 @@ console.log("lastIndiceslastIndices",lastIndices)
...
+      name: attributeName,
+      type: selectedDataType,
+    };
+    lastIndices[lastIndices.length - 1] = {
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: use condition operator (?)

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 9]
```diff
@@ -131,28 +143,40 @@ console.log("lastIndiceslastIndices",lastIndices)
...
 
-  const handleRowClick = (node:any) => {
-    console.log("Getting innnnnn");
+  const handleRowClick = (node: any) => {
     setSelectedNodeDetails([node]);
-    handleAddAttributeDetails(lastIndices);
-    console.log("Data getting printed on clicking the attribute", lastIndices);
-    console.log("selectedNodeDetails", selectedNodeDetails);
-    console.log("Getting outttttttt");
+
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx:160]: use condition operator (?) and remove commented lines

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 10]
```diff
@@ -264,23 +311,28 @@ console.log("lastIndiceslastIndices",lastIndices)
...
-                  onClick={() => handleRowClick(node.metadata)}
+                  onClick={() => handleRowClick(node)}
                   style={{ cursor: "pointer" }}
                 >
                   <Col span={28}>
                     {node.name && (
                       <p>
-                         <span style={{ color: 'black', fontWeight: '500' }}>{node.name}</span>
-                        
+                        <span style={{ color: "black", fontWeight: "500" }}>
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx:318]: use condition operator (?)

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx [hunk 11]
```diff
@@ -300,10 +352,20 @@ console.log("lastIndiceslastIndices",lastIndices)
...
         ) : (
-          <GroupsThirdSide selectedNodeDetails={selectedNodeDetails} setSelectedNodeDetails={setSelectedNodeDetails}/>
+          <GroupsThirdSide
+            selectedNodeDetails={selectedNodeDetails}
+            setSelectedNodeDetails={setSelectedNodeDetails}
+            lastIndices={lastIndices}
+            setLastIndices={setLastIndices}
+          />
         )}
+        {/* {displayAttributeSection} */}
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: remove commented line

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBoxButtons.tsx [hunk 12]
```diff
@@ -42,11 +42,11 @@ const GroupsModalBoxuttons = ({handleSaveModalCancel,handleSaveModalOk,saveModal
...
   const handleDeleteModalOk = () => {
-    setDeleteModalVisible(false);
-    dispatch(deleteGroupModalBox());
     setName("");
-    setDescription("");
-    showSuccessToast("Deleted Successfully")
+  setDescription("");
+  dispatch(deleteGroupModalBox());
+  setDeleteModalVisible(false);
+  showSuccessToast("Deleted Successfully");
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsModalBoxButtons.tsx]: move hardcoded values to constant file and show toast successful message only when the success action happens

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx [hunk 13]
```diff
@@ -1,56 +1,102 @@
...
+          }
+        };
+      } else {
+        return node;
+      }
+    });
+
+    // Update the selected node directly in lastIndices
+    const nodeIndexInLastIndices = lastIndices.findIndex(
+      (selectedNode) => selectedNode.id === updatedSelectedNodeDetails[0].id
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx]: use condition operator (?)

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx [hunk 14]
```diff
@@ -1,56 +1,102 @@
...
+
+  
+  
+  // console.log(selectedNodeDetails, "--out--selectedNodeDetails--selectedNodeDetails");
+  
+  /////////////////////////////////////////////////////////////////////////
 
-  const conList = [{id: Math.random(),
-    component : <ConditionalDataSelection key={Math.random()}/>}]
+
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx:48]: remove commented lines..

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx [hunk 15]
```diff
@@ -186,7 +240,7 @@ const GroupsThirdSide: React.FC<MyModalProps> = ({
               </Row>
             </Col>
           </Row>
-          <Row style={{ marginTop: "1rem" }}>
+          {/* <Row style={{ marginTop: "1rem" }}>
```
Comment (MerinRaj-212052) [inline src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx]: remove commented code

File: src/components/homeLeftArea/homeLeftArea.tsx [hunk 16]
```diff
@@ -14,6 +14,7 @@ import { getProjectsSelector } from "@/redux/selector";
 const HomeLeftArea: React.FunctionComponent = () => {
   const dispatch = useDispatch();
   const projectListData = useSelector(getProjectsSelector);
+  console.log(projectListData,"-----projectListData")
```
Comment (MerinRaj-212052) [inline src/components/homeLeftArea/homeLeftArea.tsx]: remove console

File: src/pages/compose/index.tsx [hunk 17]
```diff
@@ -36,18 +37,25 @@ const Compose = () => {
   const [descriptionError, setDescriptionError] = useState(false);
   const [deleteModalVisible, setDeleteModalVisible] = useState(false);
   const [projectId, setProjectId] = useState(1);
+  const [saveClicked, setSaveClicked] = useState(false);
+  const [savedData, setSavedData] = useState(null);
+  const dispatch = useDispatch();
+
+  //useSelector to take posted - compose page name and desc
+  const postComposeNameDescData = useSelector(postComposeNameDescSelector);
```
Comment (MerinRaj-212052) [inline src/pages/compose/index.tsx:45]: remove consoles

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 3 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 4 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 5 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 6 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 7 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 8 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 9 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx:160]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 10 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx:318]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx (hunk 11 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBox.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsModalBoxButtons.tsx (hunk 12 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsModalBoxButtons.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx (hunk 13 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx (hunk 14 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx:48]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx (hunk 15 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/groupsThirdSide.tsx]: Done

File: src/components/homeLeftArea/homeLeftArea.tsx (hunk 16 above)
Comment (212040-AnnMariaJenson) [reply src/components/homeLeftArea/homeLeftArea.tsx]: Done

File: src/pages/compose/index.tsx (hunk 17 above)
Comment (212040-AnnMariaJenson) [reply src/pages/compose/index.tsx:45]: Done

File: src/components/ComposePage/GroupsPage/ModalBox/attributeDisplayBox.tsx (hunk 2 above)
Comment (212040-AnnMariaJenson) [reply src/components/ComposePage/GroupsPage/ModalBox/attributeDisplayBox.tsx:15]: Done

**Review by MerinRaj-212052 — APPROVED** [review]

File: src/components/ComposePage/CreatePipeline/BottomSection.tsx [hunk 18]
```diff
@@ -10,7 +10,7 @@ const BottomSection: React.FC<BottomSectionProps> = ({ cardSelected }) => {
 
   return (
     <>
-      <Row justify="center" align="middle">
+      <Row type="flex" justify="center" align="middle">
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/BottomSection.tsx:13]: Type flex is not supported in Row. Check the eslint error

File: src/components/ComposePage/CreatePipeline/CardDisplay.tsx [hunk 19]
```diff
@@ -1,14 +1,8 @@
...
-interface CardDisplayProps {
-  alt: string;
-  src: string;
-  title: string;
-  onSelectCard: () => void;
-  isThisCardSelected: boolean;
-}
-const CardDisplay: React.FC<CardDisplayProps> = ({ alt, src, title, onSelectCard, isThisCardSelected }) => {
+
+const CardDisplay = ({ alt, src, title, onSelectCard, isThisCardSelected }) => {
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/CardDisplay.tsx:5]: Es lint error

File: src/components/ComposePage/CreatePipeline/FloatNumberInput.tsx [hunk 20]
```diff
@@ -1,58 +1,56 @@
...
-  type?: string;
-  required?: boolean;
-  dataValue?: number;
-  onChange?: (value: number | null) => void;
-  min?: number;
-  max?: number;
-}
-const FloatNumberInput: React.FC<FloatNumberInputProps> = (props) => {
+
+const FloatNumberInput = (props) => {
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/FloatNumberInput.tsx:6]: Es lint error

File: src/components/ComposePage/CreatePipeline/FloatNumberInput.tsx [hunk 21]
```diff
@@ -1,58 +1,56 @@
...
+
+  const labelClass = isOccupied ? `${styles.label} ${styles.aslabel}` : `${styles.label} ${styles.asplaceholder}`
+
   const requiredMark = required ? <span className="text-danger">*</span> : null;
-  const handleBlur = (e: ChangeEvent<HTMLInputElement>) => {
-    console.log("value:::", value);
-    setFocus(false);
+
+  const handleBlur = e => {
+    console.log("value:::",value)
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/FloatNumberInput.tsx:21]: console needs to be removed

File: src/components/ComposePage/CreatePipeline/IntimationListInputBox.tsx [hunk 22]
```diff
@@ -45,7 +45,7 @@ const IntimationListInputBox = ({ tags, setTags, label, placeholder }: Props) =>
 
   const isOccupied = (focus || (value && value.length !== 0)) && tags.length !== 0 ;
 
-  const handleBlur =  (e: React.FocusEvent<HTMLInputElement>) =>  {
+  const handleBlur = e => {
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/IntimationListInputBox.tsx:48]: es lint

File: src/components/ComposePage/CreatePipeline/TickButton.tsx [hunk 23]
```diff
@@ -5,7 +5,7 @@ import { CheckOutlined } from '@ant-design/icons';
 import styles from '../buttons/buttonStyle.module.css';
 
 interface MyComponentProps {
-    type: "link" | "text" | "dashed" | "ghost" | "default" | "primary" | undefined;
+    type: string
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/TickButton.tsx:8]: why other types are removed?

File: src/components/ComposePage/CreatePipeline/UpperSection.tsx [hunk 24]
```diff
@@ -5,12 +5,7 @@ import { PlayCircleFilled } from "@ant-design/icons";
 import CardDisplay from './CardDisplay';
 import styles from '../buttons/buttonStyle.module.css'
 
-interface UpperSectionProps {
-    onSelectInitial: () => void;
-    onSelectSync: () => void;
-    cardSelected: string;
-  }
-const UpperSection: React.FC<UpperSectionProps>= ({ onSelectInitial, onSelectSync, cardSelected }) => {
+const UpperSection = ({ onSelectInitial, onSelectSync, cardSelected }) => {
```
Comment (sourcecoursepoc) [inline src/components/ComposePage/CreatePipeline/UpperSection.tsx:8]: Es lint. Teke lates from main

Comment (sourcecoursepoc) [reply src/components/ComposePage/CreatePipeline/UpperSection.tsx:8]: Many codes are getting overriden

**Review by sourcecoursepoc — CHANGES_REQUESTED** [review]

**Comment by sourcecoursepoc** [comment]
Not relevant

**Pull request closed without merging**